python_ta/transforms/type_inference_visitor.py

```diff
@@ -220,19 +220,27 @@ def _assign_type(self, target: NodeNG, expr_type: type) -> None:
         if isinstance(target, astroid.AssignName):
             # A single identifier, e.g. x = ...
             target_type_var = self.lookup_type(target, target.name)
-            self.type_constraints.unify(target_type_var, expr_type)
+            target.inf_type = self.type_constraints.unify(target_type_var, expr_type)
         elif isinstance(target, astroid.AssignAttr):
             # Attribute mutation, e.g. x.y = ...
             attr_type = self._lookup_attribute_type(target, target.expr.name, target.attrname)
-            self.type_constraints.unify(attr_type, expr_type)
+            target.inf_type = self.type_constraints.unify(attr_type, expr_type)
         elif isinstance(target, astroid.Tuple):
             # Unpacking assignment, e.g. x, y = ...
             if isinstance(expr_type, typing.TupleMeta):
-                # TODO: handle when these collections are different lengths.
-                self.type_constraints.unify(
-                    Tuple[tuple(self.lookup_type(subtarget, subtarget.name) for subtarget in target.elts)],
-                    expr_type
-                )
+                if len(expr_type.__args__) == len(target.elts):
+                    target.inf_type = self.type_constraints.unify(
+                        Tuple[tuple(self.lookup_type(subtarget, subtarget.name) for subtarget in target.elts)],
+                        expr_type
+                    )
+                    for subtarget, subvalue in zip(target.elts, expr_type.__args__):
+                        subtarget_type = self.lookup_type(subtarget, subtarget.name)
+                        subtarget.inf_type = self.type_constraints.unify(subtarget_type, subvalue)
+                else:
+                    # TODO: Change error message to include more information
+                    target.inf_type = TypeFail("Wrong number of arguments when assigning to Tuple")
```

Comment:
This is the right idea, but it should be the `Assign` node itself that gets the error.